Updated devOptions page for newer format

LiamRotchford5671 opened the pull request:
New format updated in devOptions.html from tabs to a registration form
Adds easier / faster input for making dev choices on the state variables 
Added default fill in of data for the text fields and radio buttons as the default play values to make input choices faster
Fully tested all US-2 story criteria with master branch code to confirm everything is still working properly Done-Done

This newest devOptions.html is ready to be pushed into master.

LiamRotchford5671 committed:
Updated devOptions page for newer format

New format updated in devOptions.html from tabs to a registration form
Adds easier / faster input for making dev choices on the state variables 
Added default fill in of data for the text fields and radio buttons as the default play values to make input choices faster
Fully tested all US-2 story criteria with code to confirm everything is still working properly Done-Done

halyngoc reviewed (dismissed):
- [x] [Line 65](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/be4a82da57318051ad500385b784e92a5683e729/devOptions.html#L65) has an extra `<input `
- [x] [Line 80](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/be4a82da57318051ad500385b784e92a5683e729/devOptions.html#L80) and [87](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/be4a82da57318051ad500385b784e92a5683e729/devOptions.html#L87) `checked` is enough, no need for `checked="checked"` (see [tutorial](https://www.w3schools.com/tags/att_input_checked.asp))
- [x] You should put everything in `<style>` in a separate css file and link it back to the html file with `<link>`. Although it's find if you don't do this.

LiamRotchford5671 commented:
Fixed the first two changes requested, attempted the third but am having a issue getting the link tag to work and use the separated .css file. Will work on this further later on, for now I'll commit the working version without the separation change request.

LiamRotchford5671 committed:
Fixed changes request in PR 

Fixed extra <input on line 65
Changed checked="checked" tags on the radio buttons to just checked

halyngoc commented on devOptions.html:
```diff
@@ -1,238 +1,154 @@
-    <input type="text" name="energy" id="energy" value=""><br><br>
-</div>
+    <br><br>
+    <a type="submit" class="SubmitData" href="SpaceHunt.html" target="_self" onclick="setStateData();">Submit Data</a> 
```
I don't think `class="SubmitData"` makes a lot of sense here since this button is the only thing with this class. How bout `id="devSubmitBtn"` or something like that instead?

LiamRotchford5671 replied:
> I don't think `class="SubmitData"` makes a lot of sense here since this button is the only thing with this class. How bout `id="devSubmitBtn"` or something like that instead?

The class attribute is what applies the CSS style to the button, if its changed to id it changes the button to a very out of place looking link so I think class is fine. However, I did change the name to devSubmitBtn since it is more clear.

halyngoc replied:
Well you can select by id like this `#devSubmitBtn` in CSS

LiamRotchford5671 replied:
I just figured that out as well, changes have been made however, I still can't get the link tag to work with the now external CSS file.

LiamRotchford5671 replied:
Never mind, just solved it all. Completing the last requested change and pushing a commit to the branch.

LiamRotchford5671 committed:
Implemented linked external CSS style file 

Implemented link tag for external CSS stylesheet file for the devOptions.html
Also changed class attribute for submit button and div tag for the form to use the id attribute instead.

LiamRotchford5671 committed:
Updated devOptions to have external js file

Remove the internal js functions from the devOptions html file

This will result in easier readability and organization

As a result we now have the devDataSave.js file that it connected and utilized by the devOptions.html file

SimonBartonPSU commented:
Nice updates. New layout should make testing easier.

Is [this alert](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/cf00b67a521cfa5c391330d7a19b608172e52575/devDataSave.js#L47) just for double checking values before returning to SpaceHunt.html?

LiamRotchford5671 commented:
> Nice updates. New layout should make testing easier.
> 
> Is [this alert](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/cf00b67a521cfa5c391330d7a19b608172e52575/devDataSave.js#L47) just for double checking values before returning to SpaceHunt.html?

I forgot to comment that out before committing because i was rushing before my laptop died but yes. I just got home and will be working on the hound bot stuff in the new .js file shortly.

LiamRotchford5671 committed:
Hound Bot edits for Out of bounds vars

Fixed the minor errors mentioned by the Bot for out of scope vars / double variable names.

LiamRotchford5671 commented:
> > Nice updates. New layout should make testing easier.
> > Is [this alert](https://github.com/halyngoc/CS300-Group3-SpaceHunt/blob/cf00b67a521cfa5c391330d7a19b608172e52575/devDataSave.js#L47) just for double checking values before returning to SpaceHunt.html?
> 
> I forgot to comment that out before committing because i was rushing before my laptop died but yes. I just got home and will be working on the hound bot stuff in the new .js file shortly.

Fully finished all hound bot fixes, everything is working properly and should be ready for final review and push.

LiamRotchford5671 committed:
Merge branch 'master' into DevOptionsPatch

SimonBartonPSU reviewed (approved):
Looks great. With no conflicts, I'd say let's merge!
